web/src/compose_validate.ts

```diff
@@ -110,6 +112,46 @@ export function needs_subscribe_warning(user_id: number, stream_id: number): boo
     return true;
 }
 
+export function check_dm_permissions_and_get_error_string(user_ids_string: string): string {
+    if (!people.user_can_direct_message(user_ids_string)) {
+        const {name} = get_user_group_from_id(realm.realm_direct_message_permission_group);
+        if (name === "role:nobody") {
+            return $t({
+                defaultMessage: "You are not allowed to send direct messages in this organization.",
+            });
+        }
+        const display_name = settings_config.system_user_groups_list.find(
+            (group) => group.name === name,
+        )?.display_name;
+        if (display_name) {
+            return $t(
+                {
+                    defaultMessage:
+                        "{allowed_user_group} must be in every direct message conversation.",
+                },
+                {allowed_user_group: display_name.replace(" and ", " or ")},
```

Comment:
This is not valid string manipulation for any product with internationalization. Take some time to read
https://zulip.readthedocs.io/en/latest/translating/internationalization.html, but also just think about how this could possibly work.